doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -479,9 +479,117 @@
  *
  *
- * @section symb_diff_1 Symbolic differentiation
- *
- * TODO. As a temporary placeholder, here is a link to the Wikipedia article on
- * <a href="https://en.wikipedia.org/wiki/Symbolic_differentiation">symbolic differentiation</a> and the
- * <a href="https://www.sympy.org/en/index.html">SymPy</a> library.
+ * @section symb_diff_1 Symbolic expressions and differentiation
+ *
+ * <a href="https://en.wikipedia.org/wiki/Symbolic_differentiation">Symbolic differentiation</a> is,
+ * in terms of its design and usage, quite different to automatic differentiation.
+ * Underlying any symbolic library is a computer algebra system (CAS) that implements a
+ * language and collection of algorithms to manipulate symbolic (or "string-like") expressions.
+ * This is most similar, from a philosophical point of view, to how algebraic operations would be
+ * performed by hand.
+ *
+ * To help better distinguish between symbolic differentiation and numerical methods like automatic
+ * differentiation, let's consider a very simple example.
+ * Suppose that the function $f(x,y) = (2x+1)^{y}$, where $x$ and $y$ are variables that are independent
+ * of one another.
+ * By applying the chain-rule, the derivatives of this function are simply
+ * $\dfrac{d f(x,y)}{d x} = 2y(2x+1)^{y-1}$ and
+ * $\dfrac{d f(x,y)}{d y} = (2x+1)^{y} \ln(2x+1)$.
+ * These are exactly the results that you get from a CAS after defining the symbolic variables
+ * `x` and `y`, defining the symbolic expression `f = pow(2x+1, y)` and computing the
+ * derivatives `diff(f, x)` and `diff(f, y)`.
+ * At this point there is no assumption of what `x` and `y` represent; they may later interpreted as
+ * plain (scalar) numbers, complex numbers, or something else for which the power and natural
+ * logarithm functions are well defined.
+ * Obviously this means that there is also no assumption about which point to evaluate either
+ * the expression or its derivatives.
+ * One could readily take the expression for $\dfrac{d f(x, y)}{d x}$ and evaluate it
+ * at $x=1, y=2.5$ and then later, with no recomputation of the derivative expression itself,
+ * evaluate it at $x=3.25, y=-6$.
+ * In fact, the interpretation of any symbolic variable or expression, and the inter-dependencies
+ * between variables, may be defined or redefined at any point during their manipulation;
+ * this leads to a degree of flexibility in computations that cannot be matched by
+ * auto-differentiation.
+ * For example, one could perform a the permanent substitution
+ * $g(x) = \dfrac{d f(x, y)}{d x} \vert_{y=1}$ and then recompute
+ * $g(x)$ for several different values of $x$.
+ * One could also post-factum express an interdependency between `x` and `y`, such as
+ * $y \rightarrow y(x) := 2x$.
+ * For such a case, this means that the initially computed derivatives
+ * $\dfrac{d f(x, y)}{d x} \rightarrow \dfrac{\partial f(x, y(x))}{\partial x} = 2y(2x+1)^{y(x)-1} = 2y(2x+1)^{2x-1}$ and
```

Comment:
Shouldn't it be `4x(2x+1)^{2x-1}` instead of `2y(2x+1)^{2x-1}`?

Reply:
Indeed :-D